src/controller/maze2d.java

```diff
@@ -0,0 +1,138 @@
+package controller;
+
+import java.awt.*;
+import java.util.*;
+
+public class maze2d {
+
+    String mazeName = null;
+    int mazeSize;
+    boolean[][] mazeStructure;
+    Point entrance;
+    Point exit;
+
+    public maze2d(String mazeName, int mazeSize, Point entrance, Point exit, boolean[][] mazeStructure) {
+
+        this.mazeName = mazeName;
+        this.mazeSize = mazeSize;
+        this.entrance = entrance;
+        this.exit = exit;
+        this.mazeStructure = mazeStructure;
+    }
+
+
+    public maze2d(int mazeSize) {
+
+        // initiating maze structure (default params are false):
+        this.mazeStructure = new boolean[mazeSize][mazeSize];
+
+        this.mazeSize = mazeSize;
+
+        // default entrance point is (0,0):
+        this.entrance = new Point(0, 0);
+
+        // default exit is the bottom right corner:
+        this.exit = new Point(mazeSize - 1, mazeSize - 1);
+    }
+
+
+    public void setMazeName(String newMazeName) {
+        this.mazeName = newMazeName;
+    }
+
+    public boolean[][] getMazeStructure() { return this.mazeStructure; }
+
+    public String getMazeName() {
+        return this.mazeName;
+    }
+
+    public Point getEntrance() {
+        return entrance;
+    }
+
+    public Point getExit() {
+        return exit;
+    }
+
+
+    public void setCustomMaze(boolean[][] customMaze) {
+        this.mazeStructure = customMaze;
```

Comment:
Update mazeSize

Reply:
Not clear:
-Updating the maze size using a new method? not something I wish to add. 
-setCustomMaze to get mazeSize parameter? not sure why, because I wish to able the update only for the structure of the maze, not it's size

Reply:
Second point- If you update the structure of the maze, you probably need to update it's size (but for sure have to check if it's different)
The solution is to just update the size (In case there was no size difference it won't matter)

Reply:
I added new exception in case one of the structure lengths isn't equal to mazeSize.